File: Chapter_1/dualpal/dualpal.java

/*
ID: 100021881
LANG: JAVA
PROG: dualpal
*/

import java.io.*;
import java.util.*;

public class dualpal {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new FileReader("dualpal.in"));
        PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("dualpal.out")));
        String l = br.readLine();
        StringTokenizer line = new StringTokenizer(l);
        int N = Integer.parseInt(line.nextToken());
        int S = Integer.parseInt(line.nextToken());
        int count = 0;
        for (int i = S + 1; count < N; i++) {
            int numBases = 0;
            for (int base = 2; base <= 10; base++) {
                String result = Integer.toString(i, base);
                String reverse = new StringBuilder(result).reverse().toString();
                if (reverse.equals(result)) {
                    numBases++;
                }
            }
            if (numBases >= 2) {
                count++;
                out.println(i);
            }
        }
        out.close();
        br.close();
    }
}

File: Chapter_1/dualpal/dualpal.cpp
/*
ID: 100021881
LANG: C++
PROG: dualpal
*/
#include <iostream>
#include <fstream>
#include <string>
#include <algorithm>

using namespace std;

int main() {
    ifstream fin("dualpal.in");
    ofstream fout("dualpal.out");
    int N, S;
    fin >> N >> S;
    int count = 0;
    for (int i = S+1; count < N; i++) {
        int numBases = 0;
        for (int base = 2; base <= 10; base++) {
            char buffer[33];
            itoa(i, buffer, base);
            string result(buffer);
            string reverse(result.rbegin(), result.rend());
            if (result==reverse) numBases++;
        }
        if (numBases>=2) {
            count++;
            fout << i <<  endl;
        }
    }
    exit(0);
}